File: api/src/main/java/org/mcphoton/plugin/PluginDescription.java
package org.mcphoton.plugin;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Annotation that must be applied on plugin classes, so that we can get the plugin's basic
 * informations without creating an instance of it.
 *
 * @author TheElectronWill
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface PluginDescription {
	String name();

	String version();

	String author();

	String[] requiredDependencies() default "";

	String[] optionalDependencies() default "";
}

File: core/src/org/tuubes/core/plugins/PluginDescription.scala
package org.tuubes.core.plugins

/**
 * Trait that must be implemented by the companion object of each plugin's main class. It allows
 * Tuubes to access some plugin's information without creating an instance of that plugin.
 *
 * @see org.tuubes.core.plugins.PluginInfos
 * @see com.electronwill.macros.PluginMain
 * @author TheElectronWill
 */
trait PluginDescription {
  val name: String
  val version: String
  val optionalDeps: Seq[String]
  val requiredDeps: Seq[String]
}
